jni/src/main/java/com/horizen/librustsidechains/FieldElement.java

```diff
@@ -12,9 +14,57 @@ private FieldElement(long fieldElementPointer) {
         this.fieldElementPointer = fieldElementPointer;
     }
 
-    private static native FieldElement nativeCreateFromLong(Long value);
+    private static native FieldElement nativeCreateFromLong(long value);
 
-    public static FieldElement createFromLong(Long value) {
+    public static FieldElement createFromLong(long value) {
         return nativeCreateFromLong(value);
     }
+
+    private static native FieldElement nativeCreateRandom();
+
+    public static FieldElement createRandom() { return nativeCreateRandom(); }
+
+    private native byte[] nativeSerializeFieldElement();
+
+    public byte[] serializeFieldElement() {
+        if (fieldElementPointer == 0)
+            throw new IllegalArgumentException("Field element was freed.");
+
+        return nativeSerializeFieldElement();
+    }
+
+    private static native long nativeDeserializeFieldElement(byte[] fieldElementBytes);
+
+    public static FieldElement deserialize(byte[] fieldElementBytes) {
+        if (fieldElementBytes.length != FIELD_ELEMENT_LENGTH)
```

Comment:
Is it really needed ? Because that would be a particular case, and these cases are already handled by JNI functions which have very specific input parameters (e.g. the createNaiveThresholdProof function that takes the block hashes as 32 byte array and then does the conversion with padding internally)

Reply:
Resolved.